### PR TITLE
Add Snow Cones and Snow Machines

### LINKED ISSUE #73809: Snow-Cones
### Is your feature request related to a problem? Please describe.

Seems like a fun idea

### Solution you would like.

Add in an appliance for a snow machine. Recipe goes as follows: 
Tools: Snow Machine
Materials: Clean water (2 units) & [Syrups & sugary drinks (I.E. Maple syrup or cola)]

### Describe alternatives you have considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TheMurderUnicorn**
#### Summary
Content "Add Snow Cones and Snow Machines"

#### Purpose of change
#73809 requested snow cones and that sounded like something fun to work on so I could learn how we contextually name things.  People like snow cones, hence PR
#### Describe the solution
Please note that while this would not be a large PR for a lot of contributors, it was a lot for me and took quite a long time.  I may have made mistakes or forgotten something glaring, if anyone sees anything wrong in the files please point it out I'd appreciate it.

<details>
<summary>Snow Machine</summary>
- Add Snow Machine: This comes in the form of a home plug-in device that freezes water and grinds it into ‘snow’ for making snowcones.  Based on: https://www.amazon.com/Hawaiian-Shaved-Ice-S900A-Machine/dp/B004UBMRKG/ref=asc_df_B004UBMRKG but tweaked slightly. Added to cooking.json, line 862ish. I had to make it use water as an input because we cannot specify components be frozen, IE: I couldn’t make it use ice only.  As a result, snowcones take a long time to craft to simulate freezing the water to make the snowcone.  I would like to add a standing appliance using the appliance system, and add another recipe using that bigger machine to create them faster, but this was not done in this PR.
</details>
<details>
<summary>Snow Making tool Quality</summary>
- Snow Making Tool Quality: Made the snow making quality, and assigned level 1 to the above snow machine. Added to data/json/tool_qualities.json, line 323 ish.  I like using a tool quality for this because if I make a second machine as I mentioned above, I could give it a higher quality and use that to separate into a second recipe that could have different time and energy consumptions, etc.
</details>
<details>
<summary>Snow Cone Item/Recipe</summary>
 -Add Snowcones/Recipe: Added snowcone, which is 2 clean water and 1 flavoring put together using the above snow machine.  I have added most drinks as possible flavoring, however, I have left out alcoholic beverages, most caffeinated beverages, and some thicker syrups/materials that would not freeze well (IE: dumping honey on ice isn’t going to make a good snowcone when it immediately crystallizes).  Snowcone added to comestibles/frozen.json line 84ish, recipe added to recipes/food/frozen.json above ice cream around line 154

 -Why Not Ice Instead of Water?: I had to make it use water as an input because we cannot specify components be frozen, IE: I couldn’t make it use ice only.  Even if I created a standalone ice item, it would be meltable, and I again wouldn’t be able to specify it to be frozen, which is the same problem, so, yeah.  It uses water.  

 -Why 10 Minutes?: I have to have some way of showing the water is being converted to ice, and so I added a 10 minute freeze time.  It is definitely not ideal, I’d love to call it 5 minutes and just handwave it, but I figured I shouldn’t do that.  I’d like to add a larger machine in commercial locations as a standalone appliance, give THAT a higher snow making quality, and duplicate the recipe to use that quality instead.  Then I could drop the time significantly and just say it’s a very efficient machine compared to the home version, but that’s a separate PR if I do it at all.  Any thoughts on this?

 -Why Melted?: You cannot craft items frozen, or at least no way that I found and no one else seemed to know how to do it.  So, it spawns melted and must be frozen manually, which is also the case for things like milk shakes and ice cream/sorbet/whatever.  So it’s within the bounds of how other things are handled, it just isn’t ideal.

 -Contextual Names for Snowcones: I tried to add a contextual name for most of the flavorings, IE: a grape juice flavored snowcone will produce a “grape snowcone”.  However, some of these have issues, specifically apple.  The ID for apples is just ‘apple’, but to name it as an apple snowcone, it would catch ALL things that contain the word apple, meaning apple cider, crabapple, pineapple, etc. would all be renamed to “apple snowcone”, which we don’t want.  To get around this, apple snowcones simple produce “snowcone” items, with no contextual name.  It’s not ideal.

 -Made Snowcones Liquid: Unfortunately, this is not an ideal solution.  Because they’re liquid, they can be poured into containers that would not normally contain a snowcone, IE: you wouldn’t be eating a snowcone out of a canteen.  However, the alternative was having you carry snowcones barehanded outside of containers.  This seemed the more appropriate situation.
</details>

<details>
<summary>Snow Machine Spawns</summary>
-Add Snowmachines to Residential Spawns: Since the device is designed specifically for home use, it should really only appear in residential locations.  I’m not saying that it would be impossible to find in a diner or whatever, it’s just more likely they’d have commercial machines, which would come in a later PR.  For simplicities sake, I’ve added them to SUS appliances and kitchen appliances group.  SUS_appliances_cupboard (line 507 of domestic.json, “prob” of 5 so 5% chance of spawning) and kitchen_appliances (line 134 in collections_domestic.json, “prob” of 5 so 2% chance of spawning)
</details>

#### Describe alternatives you've considered
Many, they were listed throughout my explanations above, here are some for this section I guess: As I said I’d like to make a second machine that produces them faster.  I debated leaving them as a solid instead of liquid but dismissed that.  I debated adding alcoholic snowcones in this PR but decided it was better in a separate PR.  I debated trying to code some sort of flag to allow them to be crafted frozen instead of melted but that’s beyond my paygrade/c++ I can’t do probably.  I discussed making it always a proper appliance, but most of the home versions are small about the size of a coffeemaker so it made more sense to treat it as a ‘power tool’ like this. 

#### Testing
Tested the recipe a lot, made sure (most, see my notes) that the contextual names produced properly, that the recipe worked, that there were no errors or syntax issues.  Tested itemgroups to ensure that the ice machine was spawning at 5% and 2% respectively (see above notes), tested everything I could think of.

#### Additional context
closes #73809 

<details>
<summary>Images</summary>
item based on: For some reason this image isn't embedding properly, but the link given still works to see the image
image

example snowcones, albeit before I made them liquid so they're not in containers:
image

</details>

Item based on: https://www.amazon.com/Hawaiian-Shaved-Ice-S900A-Machine/dp/B004UBMRKG/ref=asc_df_B004UBMRKG
Misc info: Measurements: 8”D x 8”W x 12”H = 8×8×12 =768 inches3  = 12.585 L
Package weight was listed at 3 lb, I’m sure that includes packing but let’s just call it a 3lb machine.
55 watts, 120 volts, plastic material, metal blade

If this goes well and there are no glaring issues, I plan to add alcoholic snowcones in a separate PR or at least those that are strong flavored alcohols (gin, pina colata, etc).  Since we only have two flags for alcohol strength, this is easy.  As opposed to making caffeinated snowcones, which all have their own individual positive stim values that range quite a lot.

### CODE REVIEW

File: data/json/items/tool/cooking.json [hunk 1]
```diff
@@ -860,6 +860,27 @@
     },
     "melee_damage": { "bash": 4 }
   },
+  {
+    "id": "snow_machine_home",
+    "type": "TOOL",
+    "name": { "str": "home snow machine" },
+    "description": "A snow machine, which freezes water and shreds it into \"snow\", designed to be used from the comfort of your own home!  Useful for making snowcones.",
```
Comment (Karol1223) [inline data/json/items/tool/cooking.json:867]:
```suggestion
    "description": "A snow machine, which freezes water and shreds it into \"snow\", designed to be used from the comfort of your own home!  Useful for making snow cones.",
```

You're gonna wanna add a space here to please the spellchecker.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Kamejeir** [comment]
an educated, if slightly unrefined guess as how to make you able to craft frozen stuffs:
<details>

defining a copy of this function "hot_result" into something like frozen_result while setting your snow maker to be what means to being crafted with a chilling tool in the same way
(there's clearly a better way than to redo these checks prior of checking tool type, but I'll save that for if you follow up on this) https://github.com/CleverRaven/Cataclysm-DDA/blob/140db31b8fabe28a7714187bbd7fe88766396ebb/src/recipe.cpp#L1483-L1506
then adding a should_freeze on the side of should_heat https://github.com/CleverRaven/Cataclysm-DDA/blob/140db31b8fabe28a7714187bbd7fe88766396ebb/src/crafting.cpp#L1498-L1509
afterwards, adding another if statement for that aforementioned should_freeze https://github.com/CleverRaven/Cataclysm-DDA/blob/140db31b8fabe28a7714187bbd7fe88766396ebb/src/crafting.cpp#L1451-L1466
lastly, making a "freeze_up" function akin to seen here https://github.com/CleverRaven/Cataclysm-DDA/blob/140db31b8fabe28a7714187bbd7fe88766396ebb/src/item.cpp#L13043-L13055

not forgetting adding the new functions to the respective header files:
https://github.com/CleverRaven/Cataclysm-DDA/blob/140db31b8fabe28a7714187bbd7fe88766396ebb/src/item.h#L1056-L1060
https://github.com/CleverRaven/Cataclysm-DDA/blob/140db31b8fabe28a7714187bbd7fe88766396ebb/src/recipe.h#L299

</details>

File: data/json/items/comestibles/frozen.json [hunk 2]
```diff
@@ -81,6 +81,89 @@
...
+    "quench": 40,
+    "charges": 1,
+    "fun": 6,
+    "description": "A summer treat of chopped ice and sweet flavoring!",
+    "price": "3 USD 50 cent",
+    "price_postapoc": "50 cent",
+    "material": [ "water" ],
+    "//": "had to make it a liquid, its not perfect but the alternative is barehanding loose ice soaked in liquid with no containers",
+    "phase": "liquid",
+    "flags": [ "MELTS" ],
```
[the remark this answers is not in the document]
Comment (Kamejeir) [reply data/json/items/comestibles/frozen.json:164]: a collaborator corrected me

**Comment by Maleclypse** [comment]
I thought you crafted ice cream as frozen already?

**Comment by Kamejeir** [comment]
>I thought you crafted ice cream as frozen already?

image
I'm surprised, myself. (on the master branch) Granted, this was done using Debug Hammerspace, if that is somehow relevant.

**Comment by I-am-Erk** [comment]
I'm trying to think of a joke to put here but I'm just happy to have ice cream.

**Pull request merged**